assets/src/js/query-no-posts.js

```diff
@@ -2,8 +2,10 @@ export const removeNoPostText = () => {
   const postTitle = document.querySelector('.p4-query-loop .wp-block-heading');
   const postDescription = document.querySelector('.p4-query-loop p');
 
-  if (postTitle.innerHTML === '' && postDescription.innerHTML === '') {
-    const noResultsText = document.querySelector('.p4-query-loop .wp-block-query-no-results');
-    noResultsText.style.display = 'none';
+  if (!postTitle || !postDescription || postTitle.innerHTML !== '' || postDescription.innerHTML !== '') {
+    return;
   }
+
+  const noResultsText = document.querySelector('.p4-query-loop .wp-block-query-no-results');
+  noResultsText.style.display = 'none';
```

Comment:
Should we check that `noResultsText` is not `undefined`?
```suggestion
  if(noResultsText) noResultsText.style.display = 'none';
```

Reply:
You have a good point 🤔 It should always be there in our Query block variations, but editors can make their own!